d3a_interface/utils.py

```diff
@@ -185,7 +185,14 @@ def str_to_pendulum_datetime(input_str):
         try:
             pendulum_time = from_format(input_str, DATE_TIME_FORMAT)
         except ValueError:
-            raise Exception(f"Format is not one of ('{TIME_FORMAT}', '{DATE_TIME_FORMAT}')")
+            try:
+                pendulum_time = from_format(input_str, DATE_TIME_FORMAT_SECONDS)
+            except ValueError:
+                try:
+                    pendulum_time = from_format(input_str, TIME_FORMAT_SECONDS)
+                except ValueError:
+                    raise Exception(f"Format is not one of ('{TIME_FORMAT}', '{DATE_TIME_FORMAT}'"
```

Comment:
This is linearly scaling up for each new format we add, I would suggest using a different approach like the following:
```python
supported_formats = iter([TIME_FORMAT, DATE_TIME_FORMAT_SECONDS, TIME_FORMAT_SECONDS])
pendulum_time = None
for format in supported_formats:
    try:
        pendulum_time = from_format(input_str, format)
        break
    except ValueError:
        continue
if not  pendulum_time:
    raise Exception(f"Format is not ....")
```

Reply:
Agreed, it is much better this way! Fixing this